### DISCUSSION TITLE
Spam prevention

### DISCUSSION THREAD

**Discussion by damms005** [austintoddj/canvas#1088] [New Features]
Is it possible to add https://github.com/spatie/laravel-honeypot to prevent bot spamming, or is pr welcomed in this regard?

**Comment by austintoddj** [comment]
In terms of Canvas UI, I think leaving it more open to various options makes the most sense. Laravel Honeypot is a fine package for this, but I don't want to constrain developers to one specific form of spam mitigation. I appreciate the suggestion though!

**Reply by damms005** [reply]
My mistake. I wrongly assumed that Canvas has comments feature. Hence my suggestion above. I did some findings and came across  [issues #683](https://github.com/austintoddj/canvas/issues/683). What happened to Studio that you [suggested in your comment](https://github.com/austintoddj/canvas/issues/683#issuecomment-606546251)?

Secondly, is contribution welcome for this comments feature?

**Reply by austintoddj** [reply]
Canvas UI is essentially the evolved Studio. It was a standalone package that you could pull in and use to display a forward-facing blog. If I remember correctly, that did use Disqus for commenting, but it's been a while since I looked.

Now everything is consolidated into this project (optional publishing of the frontend with the `canvas:ui` Artisan command). When v7 releases, it's going to get an overhaul.

While I appreciate that you want to contribute that, I don't want to bake in a commenting system in general and instead let users do that themselves with the published files. Canvas UI's purpose is very simply: display the posts/tags/topics that have been published. From there, developers can extend in whatever way suites their needs.

**Reply by damms005** [reply]
Makes sense. Thanks for all you do man

**Comment by austintoddj** [comment]
Closing as obsolete with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).
